hazelcast-wm/src/main/java/com/hazelcast/web/ClusteredSessionService.java

```diff
@@ -0,0 +1,270 @@
+    private volatile boolean failedConnection = true;
+    private volatile long lastConnectionTry;
+    private final ScheduledExecutorService es = Executors.newSingleThreadScheduledExecutor(new EnsureInstanceThreadFactory());
+    private volatile boolean ensureInstanceRunning = true;
+
+
+    public ClusteredSessionService(FilterConfig filterConfig, Properties properties, String clusterMapName, String sessionTTL) {
+        this.filterConfig = filterConfig;
+        this.properties = properties;
+        this.clusterMapName = clusterMapName;
+        this.sessionTTL = sessionTTL;
+
+        es.scheduleWithFixedDelay(new Runnable() {
+            public void run() {
+                while (ensureInstanceRunning) {
+                    try {
+                        ensureInstance();
+                    } catch (Exception ignored) {
+                        if (ignored instanceof InterruptedException) {
+                            ensureInstanceRunning = false;
+                        }
+                        EmptyStatement.ignore(ignored);
+                    }
+                }
+            }
+        }, 0, CLUSTER_CHECK_INTERVAL, TimeUnit.SECONDS);
+        try {
+            ensureInstance();
+        } catch (Exception ignored) {
+            EmptyStatement.ignore(ignored);
+        }
+    }
+
+    private void ensureInstance() throws Exception {
+        if (failedConnection && System.currentTimeMillis() > lastConnectionTry + RETRY_MILLIS) {
+            synchronized (this) {
+                try {
+                    if (failedConnection && System.currentTimeMillis() > lastConnectionTry + RETRY_MILLIS) {
+                        LOGGER.info("Retrying the connection!!");
+                        lastConnectionTry = System.currentTimeMillis();
+                        hazelcastInstance = HazelcastInstanceLoader.createInstance(filterConfig, properties);
+                        clusterMap = hazelcastInstance.getMap(clusterMapName);
+                        sss = (SerializationServiceSupport) hazelcastInstance;
+                        try {
+                            if (sessionTTL != null) {
+                                Config hzConfig = hazelcastInstance.getConfig();
+                                MapConfig mapConfig = hzConfig.getMapConfig(clusterMapName);
+                                mapConfig.setTimeToLiveSeconds(Integer.parseInt(sessionTTL));
+                                hzConfig.addMapConfig(mapConfig);
+                            }
+                        } catch (UnsupportedOperationException ignored) {
+                            LOGGER.info("client cannot access Config.");
+                        }
+                        failedConnection = false;
+                        LOGGER.info("Successfully Connected!");
+                        String sessionId = orphanSessions.poll();
+                        while (sessionId != null) {
+                            if (!deleteSession(sessionId, false)) {
```

Comment:
Good catch, I fixed it `orphanSessions` now stores `SimpleEntry<String,Boolean>` instead of `String` and `deleteSession` is called with actual value of `invalidate`.